lib/rabbitmq/cli/ctl/commands/encode_command.ex

```diff
@@ -0,0 +1,135 @@
+  def validate(args, %{list_ciphers: false, list_hashes: false}) when length(args) !== 2 do
+      {:validation_failure,
+       {:bad_argument, "Please provide a value to encode/decode and a passphrase."}}
+  end
+  def validate(args, %{list_ciphers: false, list_hashes: false} = opts) when length(args) === 2 do
+    case {supports_cipher(opts.cipher), supports_hash(opts.hash), opts.iterations > 0} do
+      {false, _, _}      -> {:validation_failure, {:bad_argument, "The requested cipher is not supported."}}
+      {_, false, _}      -> {:validation_failure, {:bad_argument, "The requested hash is not supported"}}
+      {_, _, false}      -> {:validation_failure, {:bad_argument, "The requested number of iterations is incorrect"}}
+      {true, true, true} -> :ok
+    end
+  end
+  def validate(_, _), do: :ok
+  def switches() do
+      [
+        list_ciphers: :boolean,
+        list_hashes: :boolean,
+        cipher: :atom,
+        hash: :atom,
+        iterations: :integer,
+        decode: :boolean
+      ]
+    end
+  def aliases(), do: []
+
+  def scopes(), do: [:ctl]
+
+  def run(_, %{list_ciphers: true}) do
+    {:ok, :rabbit_pbe.supported_ciphers()}
+  end
+
+  def run(_, %{list_hashes: true}) do
+    {:ok, :rabbit_pbe.supported_hashes()}
+  end
+
+  def run([value, passphrase], %{list_ciphers: false, list_hashes: false, decode: false} = opts) do
+    %{:cipher => cipher, :hash => hash, :iterations => iterations} = opts
+    try do
+      term_value = evaluate_input_as_term(value)
+      result = :rabbit_pbe.encrypt_term(cipher, hash, iterations, passphrase, term_value)
+      {:ok, {:encrypted, result}}
+    catch _, _ ->
+      {:error, "Error during cipher operation."}
+    end
+  end
+
+  def run([value, passphrase], %{list_ciphers: false, list_hashes: false, decode: true} = opts) do
+    %{:cipher => cipher, :hash => hash, :iterations => iterations} = opts
+    try do
+      term_value = evaluate_input_as_term(value)
+      term_to_decrypt = case term_value do
+        {:encrypted, encrypted_term} -> encrypted_term
+        _                            -> term_value
+      end
+      result = :rabbit_pbe.decrypt_term(cipher, hash, iterations, passphrase, term_to_decrypt)
+      {:ok, result}
+    catch _, _ ->
+      {:error, "Error during cipher operation."}
+    end
+  end
+
+  def run(_, _) do
+    {:ok, "Incorrect usage of the encode command."}
+  end
+
+  # skip the default output (otherwise, formatting issue with Erlang lists/strings
+  # which are then treated as Elixir streams and get displayed on several lines
+  def output(result, _) do
+    case result do
+      {:error, _} -> RabbitMQ.CLI.DefaultOutput
+      {:ok, _}    -> result
+    end
+  end
+
+  def formatter(), do: RabbitMQ.CLI.Formatters.Erlang
+
+  def usage, do: "[--decode] [value] [passphrase] [--list-ciphers] [--list-hashes] [--cipher cipher] [--hash hash] [--iterations iterations]"
+
+  def flags, do: Keyword.keys(switches())
+
+  def banner(_, _), do: ""
```

Comment:
There are several uses of this command, so it's better to print what it's going to do.